### PR TITLE
add partition capacity reservations

### PR DESCRIPTION
**Pull request by mwennrich**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,66 +3,64 @@ module github.com/metal-stack/metal-metrics-exporter
 go 1.21
```
Comment (majst01) [inline go.mod:3]:
```suggestion
go 1.22
```

File: metalCollector.go [hunk 2]
```diff
@@ -186,22 +203,33 @@ func (collector *metalCollector) Collect(ch chan<- prometheus.Metric) {
 
 		// {"networkId", "name", "projectId", "description", "partition", "vrf", "prefixes", "destPrefixes", "parentNetworkID", "isPrivateSuper", "useNat", "isUnderlay"}, nil,
 		ch <- prometheus.MustNewConstMetric(collector.networkInfo, prometheus.GaugeValue, 1.0, *n.ID, n.Name, n.Projectid, n.Description, n.Partitionid, vrf, prefixes, destPrefixes, n.Parentnetworkid, privateSuper, nat, underlay)
-		ch <- prometheus.MustNewConstMetric(collector.usedIps, prometheus.GaugeValue, float64(*n.Usage.UsedIps), *n.ID)
-		ch <- prometheus.MustNewConstMetric(collector.availableIps, prometheus.GaugeValue, float64(*n.Usage.AvailableIps), *n.ID)
-		ch <- prometheus.MustNewConstMetric(collector.usedPrefixes, prometheus.GaugeValue, float64(*n.Usage.UsedPrefixes), *n.ID)
-		ch <- prometheus.MustNewConstMetric(collector.availablePrefixes, prometheus.GaugeValue, float64(*n.Usage.AvailablePrefixes), *n.ID)
+		if n.Usage == nil {
+			continue
+		}
```
Comment (majst01) [inline metalCollector.go:208]: can be put to line 194 as well

**Review by majst01 — APPROVED** [review]

File: metalCollector.go (hunk 2 above)
Comment (mwennrich) [reply metalCollector.go:208]: collector.networkInfo could still be provided even if usage is missing.

**Comment by mwennrich** [comment]
will be replaced

**Pull request closed without merging**